Elena_Poleschuk/Module 3 -  Java/Week 4/Hometask6/src/by/epam/tat/lecture2/task1/main/Runner.java

```diff
@@ -0,0 +1,211 @@
+								answer = Communicator.getFlag();	
+								Communicator.stringScanner(); // Consume newline left-over
+								Communicator.out("Enter name");
+								sweetName = Communicator.stringScanner();
+								Communicator.out("Enter producer");
+								sweetProducer = Communicator.stringScanner();
+								Communicator.out("Enter price");
+								price = Communicator.intScanner();
+								Communicator.out("Enter weight");
+								weight = Communicator.intScanner();
+								if (answer == true){
+									Communicator.out("Enter \"Y\" if the sweet with stick. Press any other kay if not");
+									if (Communicator.getFlag()){
+										lollipop = true;
+									} else {
+										lollipop = false;
+									}
+									Communicator.stringScanner(); // Consume newline left-over
+									Communicator.out("Enter flavour of candy");
+									flavour = Communicator.stringScanner();
+									gift.addSweet(new SugarCandy(sweetName, sweetProducer, price, weight, lollipop, flavour));	
+									} else {
+										gift.addSweet(new ChocolateCandy(sweetName, sweetProducer, price, weight));
+									}
+								Communicator.out("________________\nPress \"Y\" to add a new sweet. Press any other kay to stop");
+								addingContinue = Communicator.getFlag();
+								}
+							}catch (ExceedCountSweetsExeption e){
+								Communicator.out(e.getMessage());
+								addingContinue = false;
+							}
+						}
+					break;
+				case 2:
+					Communicator.out("--View Sweets in the Gift--");
+					gift.printGiftInfo();
+					 
+					break;			
+				case 3:
+					Communicator.out("--Calculate Weight of the Gift--");
+					Communicator.out("Weight of the gift: " + gift.getSumWeight() + "g.");
+					break;	
+				case 4:
+					Communicator.out("--Sort Sweets in the Gift--");
+					Communicator.out("Select type of sorting press number: ");
+					Communicator.out("1. by Name");
+					Communicator.out("2. by Price");
+					Communicator.out("3. by Weigth");
+					switch (Communicator.selectAction(3))
+					{
+						case 1:
+							gift.sortByName();
+							break;
+						case 2:
+							gift.sortByPrice();
+							break;			
+						default:
+							gift.sortByWeight();	
+					};
+					break;		
+				case 5:
+					Communicator.out("--Find sweet--");
+					Communicator.stringScanner(); // Consume newline left-over
+					Communicator.out("Enter a name of sweet to find other information about it:");
+					gift.giveSweetByName(Communicator.stringScanner());
+					break;
+					
+				case 6:
+					Communicator.out("--Save gift into file--");
+					FileReaderSaver saveGiftToFile = new FileReaderSaver();
+					saveGiftToFile.saveGift(gift);
+					Communicator.out("The gift is saved");	
+					break;
+				case 7:   // open form 1.file, 2. from DB, 3. from XML
+					Communicator.out("--Open saved gift--\nSelect source of the gift.");
+					Communicator.out("1. from txt file.");
+					Communicator.out("2. from database.");
+					Communicator.out("3. from xml file.");
```

Comment:
Implementation of switch case and further functionality should looks like:

ISaver saver = null;
switch(int action)
{
 case 1:
         saver = new FileSaver();
 case 2:
         saver = new DBSaver();
 case 3:
         saver = new XMLSaver();
}
try
{
   gift.setSweets(saver.getSavedGift());
}
catch(OpeningGiftException e)
{
   Communicator.out(e.getMessage());
}

All creation of DB connection, handling of exception should be in concrete Saver implementation